tests/test_commands.py

```diff
@@ -321,74 +303,53 @@ def test_runspider_log_short_names(self):
         self.assertNotIn("[scrapy]", log2)
         self.assertIn("[scrapy.core.engine]", log2)
 
-        if sys.platform == 'win32':
-            log1 = self.get_log(self.debug_log_spider,
-                                name='myspider.pyw',
-                                args=('-s', 'LOG_SHORT_NAMES=1'))
-            self.assertIn("[myspider] DEBUG: It Works!", log1)
-            self.assertIn("[scrapy]", log1)
-            self.assertNotIn("[scrapy.core.engine]", log1)
-
-            log2 = self.get_log(self.debug_log_spider,
-                                name='myspider.pyw',
-                                args=('-s', 'LOG_SHORT_NAMES=0'))
-            self.assertIn("[myspider] DEBUG: It Works!", log2)
-            self.assertNotIn("[scrapy]", log2)
-            self.assertIn("[scrapy.core.engine]", log2)
-
     def test_runspider_no_spider_found(self):
         log = self.get_log("from scrapy.spiders import Spider\n")
         self.assertIn("No spider found in file", log)
 
-        if sys.platform == 'win32':
-            log = self.get_log("from scrapy.spiders import Spider\n", name='myspider.pyw')
-            self.assertIn("No spider found in file", log)
-
     def test_runspider_file_not_found(self):
         _, _, log = self.proc('runspider', 'some_non_existent_file')
         self.assertIn("File not found: some_non_existent_file", log)
 
     def test_runspider_unable_to_load(self):
-        log = self.get_log('', name='myspider.txt')
+        self.spider_filename = 'myspider.txt'
```

Comment:
I think it would be better not to modify `self.spider_filename` as part of this test. Maybe you can define `self.get_log` so that it only uses `self.spider_filename` if the `name` is `None` (default), so that it can accept this name as a parameter. Or maybe, since it does not make sense to run this test twice (for `.py` and for `.pyw`), maybe this test should be in a separate class, on its own.